handlers/multistage-diagram-tactile-svg/tactile_svg.py

```diff
@@ -0,0 +1,432 @@
+            arw_tgt = cohenSutherlandClip(tgt_cntr, src_cntr,
+                                          (coord_lims
+                                           [link["target"]]["lims"]))
+            arw_src = point_at_distance(arw_src, arw_tgt, 0.1)
+            arw_tgt = point_at_distance(arw_tgt, arw_src, 0.2)
+            p = draw_arrow(arw_src, arw_tgt, dimensions,
+                           label, link["directed"])
+            g.append(p)
+        except Exception as e:
+            logging.debug("Encountered error while drawing arrow")
+            logging.pii(e.message)
+
+    svg.append(g)
+
+    # Checking if graphic-caption preprocessor is present
+    if ("ca.mcgill.a11y.image.preprocessor.graphic-caption"
+            in preprocessors):
+        logging.debug("Adding title from "
+                      "graphic-caption")
+        caption = preprocessors["ca.mcgill.a11y.image."
+                                "preprocessor.graphic-caption"][
+                            "caption"]
+    else:
+        logging.debug("graphic-caption not found. "
+                      "Adding default title.")
+
+    title = draw.Title(caption)
+    svg.append(title)
+    logging.debug("Generating final rendering")
+    data = {"graphic": svg.asDataUri()}
+
+    rendering = {
+        "type_id": "ca.mcgill.a11y.image.renderer.TactileSVG",
+        "description": ("Tactile rendering of multistage diagram"),
+        "data": data
+    }
+    try:
+        validator = jsonschema.Draft7Validator(
+            renderer_schema, resolver=resolver
+        )
+        validator.validate(data)
+    except ValidationError as e:
+        logging.error("Failed to validate the response renderer")
+        logging.pii(f"Validation error: {e.message}")
+        return jsonify("Failed to validate the response renderer"), 500
+    response = {
+        "request_uuid": contents["request_uuid"],
+        "timestamp": int(time.time()),
+        "renderings": [rendering]
+    }
+    try:
+        validator = jsonschema.Draft7Validator(
+            response_schema, resolver=resolver
+        )
+        validator.validate(response)
+    except ValidationError as e:
+        logging.debug("Failed to generate a valid response")
+        logging.pii(f"Validation error: {e.message}")
+        return jsonify("Failed to generate a valid response"), 500
+    logging.debug("Sending response")
+    return response
+
+
+# Function to compute region code for a point(x, y)
+def computeCode(x, y, lims):
+    (x_min, y_min, x_max, y_max) = lims
+    code = 0
+    if x < x_min:  # to the left of rectangle
+        code |= 1
+    elif x > x_max:  # to the right of rectangle
+        code |= 2
+    if y < y_min:  # below the rectangle
+        code |= 4
+    elif y > y_max:  # above the rectangle
+        code |= 8
+    return code
+
+
+# Implementing variant of Cohen-Sutherland algorithm
+def cohenSutherlandClip(src, tgt, lims):
```

Comment:
Did you get this from anywhere?

Reply:
I did! This is a modified version of the implementation from [here](https://www.geeksforgeeks.org/line-clipping-set-1-cohen-sutherland-algorithm/)

Reply:
I would just include a comment with the source at a minimum.

Reply:
Done!